File: code/java/ch18/practice/Warmup04LemonadeChange.java
package ch18.practice;

import java.util.*;

/**
 * Warmup 4: Lemonade Change
 * Chapter 18: Greedy Algorithms — The Smart Shortcut
 *
 * PROBLEM: Each lemonade costs $5. Can you make change for everyone?
 *
 * EXAMPLES:
 *   solve([5,5,5,10,20])   -> true
 *   solve([5,5,10,10,20])  -> false
 *
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */
public class Warmup04LemonadeChange {
    public static boolean solve(int[] bills) {
        // TODO: Replace this with your solution
        return false;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int[] bills = Arrays.stream(sc.nextLine().trim().split(" ")).mapToInt(Integer::parseInt).toArray();
        System.out.println(solve(bills));
        sc.close();
    }
}

File: code/cpp/ch18/practice/warmup_04_lemonade_change.cpp
/*
 * Warmup 4: Lemonade Change
 * Chapter 18: Greedy Algorithms — The Smart Shortcut
 *
 * PROBLEM: Each lemonade costs $5. Can you make change for everyone?
 * EXAMPLES: solve({5,5,5,10,20}) -> true
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */

#include <iostream>
#include <vector>
using namespace std;

bool solve(vector<int> bills) {
    // TODO: Replace this with your solution
    return false;
}

// -- Do not change anything below this line --------------------------
int main() {
    int n; cin >> n;
    vector<int> bills(n);
    for (int i = 0; i < n; i++) cin >> bills[i];
    cout << (solve(bills) ? "true" : "false") << endl;
    return 0;
}
